exporter/signalfxexporter/exporter.go

```diff
@@ -0,0 +1,213 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package signalfxexporter
+
+import (
+	"bytes"
+	"compress/gzip"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/open-telemetry/opentelemetry-collector/consumer/consumerdata"
+	"github.com/open-telemetry/opentelemetry-collector/consumer/consumererror"
+	"github.com/open-telemetry/opentelemetry-collector/exporter"
+	"github.com/open-telemetry/opentelemetry-collector/exporter/exporterhelper"
+	sfxpb "github.com/signalfx/com_signalfx_metrics_protobuf"
+	"go.uber.org/zap"
+)
+
+// New returns a new SignalFx exporter.
+func New(
+	config *Config,
+	logger *zap.Logger,
+) (exporter.MetricsExporter, error) {
+
+	if config == nil {
+		return nil, errors.New("nil config")
+	}
+
+	if config.Name() == "" {
+		config.SetType(typeStr)
+		config.SetName(typeStr)
+	}
+
+	if config.Realm == "" {
+		err := fmt.Errorf(
+			"%q config requires a non-empty \"realm\"",
+			config.Name())
+		return nil, err
+	}
+
+	if config.Timeout == 0 {
+		config.Timeout = 5 * time.Second
+	}
+
+	if config.Timeout < 0 {
+		err := fmt.Errorf(
+			"%q config cannot have a negative \"timeout\"",
+			config.Name())
+		return nil, err
+	}
+
+	headers, err := buildHeaders(config)
+	if err != nil {
+		return nil, err
+	}
+
+	url := fmt.Sprintf("https://ingest.%s.signalfx.com/v2/datapoint", config.Realm)
+
+	s := &httpSender{
+		url:     url,
+		headers: headers,
+		client: &http.Client{
+			// TODO: What other settings of http.Client to expose via config?
```

Comment:
I'm not sure if it is on by default in Go's http.Client, but you should ensure KeepAlive/connection reuse is enabled and has a reasonably long idle timeout.

Reply:
The default is keep-alive with an idle timeout of 90 seconds.